client.go

```diff
@@ -409,40 +333,76 @@ func (r *rowWrapper) Write(dest map[string]interface{}) error {
 
 // Columns returns the column names.
 func (r *rowWrapper) Columns() ([]string, error) {
-	columns, err := r.row.Columns()
+	columns, err := r.rows.Columns()
 	if err != nil {
 		return nil, errors.Wrap(err, "makroud: cannot return row columns")
 	}
+
 	return columns, nil
 }
 
 // Scan copies the columns in the current row into the values pointed at by dest.
 // The number of values in dest must be the same as the number of columns in Rows.
 func (r *rowWrapper) Scan(dest ...interface{}) error {
-	err := r.row.Scan(dest...)
+	err := r.scan(dest...)
 	if err != nil {
 		return errors.Wrap(err, "makroud: cannot scan given values")
 	}
 	return nil
 }
 
-// A rowsWrapper wraps a rows from sqlx.
+func (r *rowWrapper) scan(dest ...interface{}) error {
+	// From sqlx source code: Discard sql.RawBytes to avoid weird issues with the SQL driver and memory management.
+	defer func() {
+		// TODO (novln): Add an observer to collect this error.
+		_ = r.rows.Close()
+	}()
+	for i := range dest {
+		_, ok := dest[i].(*sql.RawBytes)
+		if ok {
+			return errors.New("sql.RawBytes isn't allowed on Row.Scan")
+		}
+	}
+
+	if !r.rows.Next() {
+		err := r.rows.Err()
+		if err != nil {
+			return err
+		}
+		return sql.ErrNoRows
+	}
+
+	err := r.rows.Scan(dest...)
+	if err != nil {
+		return err
+	}
+
+	// Make sure the query can be processed to completion with no errors.
+	err = r.rows.Close()
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
As we don't wrap the error.

```suggestion
	return r.rows.Close()
```